In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import scale
from sklearn.decomposition import PCA


In [2]:
df_de =pd.read_excel('data.xlsx',sheet_name='sheet1',index_col=0)
df_sf6 =pd.read_excel('data.xlsx',sheet_name='sheet2')
df_oog=pd.read_excel('data.xlsx',sheet_name='sheet3')
df_ldc=pd.read_excel('data.xlsx',sheet_name='sheet4')


In [3]:
df_de.head()

FRS_Id                   Facility_Name         City State  \
Facility Id                                                                    
1004377      1.100438e+11  121 REGIONAL DISPOSAL FACILITY      MELISSA    TX   
1000112      1.100198e+11                    23rd and 3rd     BROOKLYN    NY   
1013621      1.100708e+11           3Bear Libby Gas Plant        Hobbs    NM   
1003742      1.100104e+11            31st Street Landfill  WESTCHESTER    IL   
1003188      1.100006e+11                    3M BROWNWOOD    BROWNWOOD    TX   

             Zip Code                   Address         County   Latitude  \
Facility Id                                                                 
1004377         75454  3820 SAM RAYBURN HIGHWAY  COLLIN COUNTY  33.298570   
1000112         11232            730 3rd Avenue          Kings  40.663000   
1013621         88240           674 Marathon Rd     LEA COUNTY  32.540693   
1003742         60154           11700 W 31ST ST    COOK COUNTY  41.835129   
1003188         76801    4501 HIGHWAY 377 SOUTH   BROWN COUNTY  31.671177   

              Longitude  Primary_NAICS_Code  ... Total_rep_de_2020  \
Facility Id                                  ...                     
1004377      -96.535860              562212  ...        504064.000   
1000112      -74.000000              221112  ...         60819.700   
1013621     -103.522351              211130  ...         40521.812   
1003742      -87.915924              562212  ...        100500.948   
1003188      -98.996513              326113  ...         29359.640   

            Total_rep_de_2019  Total_rep_de_2018  Total_rep_de_2017  \
Facility Id                                                           
1004377            518680.250      653854.000000      250497.500000   
1000112             44658.384       70705.288000       52698.656000   
1013621             38691.478      435280.099943      431488.809907   
1003742             18925.400       31220.150000       29026.800000   
1003188             30571.636       34432.420000       36202.264000   

             Total_rep_de_2016  Total_rep_de_2015  Total_rep_de_2014  \
Facility Id                                                            
1004377          221014.750000      199011.750000      241883.500000   
1000112           66797.352000       67399.252000       40722.200000   
1013621          447423.757318      467277.995747      491686.293592   
1003742          120657.600000       92696.600000      120707.350000   
1003188           31773.480000       36754.712000       36419.864000   

             Total_rep_de_2013  Total_rep_de_2012  Total_rep_de_2011  
Facility Id                                                           
1004377          289953.250000      204000.000000      194000.000000  
1000112          102599.220000       68984.744000      100591.828000  
1013621          500278.254166      507639.706331      534026.861412  
1003742           97467.500000       87357.750000       95113.250000  
1003188           34821.820000       33671.924000       37149.360000  

[5 rows x 22 columns]

In [5]:
df_de.columns

Index(['FRS_Id', 'Facility_Name', 'City', 'State', 'Zip Code', 'Address',
       'County', 'Latitude', 'Longitude', 'Primary_NAICS_Code', 'IT_subparts',
       'IT_sectors', 'Total_rep_de_2020', 'Total_rep_de_2019',
       'Total_rep_de_2018', 'Total_rep_de_2017', 'Total_rep_de_2016',
       'Total_rep_de_2015', 'Total_rep_de_2014', 'Total_rep_de_2013',
       'Total_rep_de_2012', 'Total_rep_de_2011'],
      dtype='object')

In [6]:
df_de_group = df_de.groupby('State')[['Total_rep_de_2020', 'Total_rep_de_2019',
       'Total_rep_de_2018', 'Total_rep_de_2017', 'Total_rep_de_2016',
       'Total_rep_de_2015', 'Total_rep_de_2014', 'Total_rep_de_2013',
       'Total_rep_de_2012', 'Total_rep_de_2011']].mean().apply(lambda x: round(x,2))
df_de_group.info()

<class 'pandas.core.frame.DataFrame'>
Index: 54 entries, AK to WY
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Total_rep_de_2020  54 non-null     float64
 1   Total_rep_de_2019  54 non-null     float64
 2   Total_rep_de_2018  54 non-null     float64
 3   Total_rep_de_2017  54 non-null     float64
 4   Total_rep_de_2016  54 non-null     float64
 5   Total_rep_de_2015  54 non-null     float64
 6   Total_rep_de_2014  54 non-null     float64
 7   Total_rep_de_2013  54 non-null     float64
 8   Total_rep_de_2012  54 non-null     float64
 9   Total_rep_de_2011  54 non-null     float64
dtypes: float64(10)
memory usage: 4.6+ KB


In [7]:
df_de_group.describe()

Total_rep_de_2020  Total_rep_de_2019  Total_rep_de_2018  \
count          54.000000          54.000000          54.000000   
mean       341519.065185      373414.029815      397565.381667   
std        160845.240990      174789.235984      186897.263877   
min         59417.180000       63455.640000       68642.990000   
25%        255241.385000      276539.762500      264171.417500   
50%        319187.815000      343344.105000      380908.440000   
75%        421860.850000      454283.352500      485810.687500   
max        776126.140000      808391.500000      878193.550000   

       Total_rep_de_2017  Total_rep_de_2016  Total_rep_de_2015  \
count          54.000000          54.000000          54.000000   
mean       397353.626111      414912.537222      436718.639630   
std        185616.317257      187268.665755      195600.449659   
min         81703.720000       85414.820000       94672.340000   
25%        263396.805000      268903.050000      290582.885000   
50%        378141.555000      397191.230000      419423.655000   
75%        452241.982500      481361.685000      516291.002500   
max        892322.330000      856710.420000      915502.800000   

       Total_rep_de_2014  Total_rep_de_2013  Total_rep_de_2012  \
count          54.000000          54.000000          54.000000   
mean       456891.439259      465164.853519      477648.179630   
std        202133.595801      201888.766538      201715.149215   
min         91641.670000       94677.980000      107202.080000   
25%        299988.907500      294338.020000      325380.107500   
50%        454315.090000      471625.920000      470430.760000   
75%        531641.140000      562387.577500      560031.542500   
max        926881.760000      989476.920000      983929.120000   

       Total_rep_de_2011  
count       5.400000e+01  
mean        5.188616e+05  
std         2.399984e+05  
min         1.173564e+05  
25%         3.648028e+05  
50%         5.099477e+05  
75%         6.063037e+05  
max         1.373103e+06

We can notice that the average direct emissions has come down from 518301 metric tons to 341519 metric tons over the decade.

In [9]:
df_de_group.sort_values(by='Total_rep_de_2020',ascending =False)

Total_rep_de_2020  Total_rep_de_2019  Total_rep_de_2018  \
State                                                            
WY             776126.14          808391.50          878193.55   
PR             696112.75          625328.24          573758.37   
FL             692818.77          694640.28          776893.27   
MO             637050.23          672098.60          732764.01   
IN             583083.48          675722.35          758857.18   
ND             580395.25          615146.86          674565.09   
WV             569000.49          664185.86          716029.37   
MT             518496.09          745868.36          706961.20   
UT             510993.04          554437.81          576151.47   
KY             494451.58          585618.34          649977.34   
AZ             487898.79          461633.61          484312.57   
TX             441232.75          482479.51          505962.69   
NV             430280.33          435262.99          434531.63   
AL             428724.98          474103.87          488460.21   
OH             401268.46          410782.36          449717.71   
MS             387816.56          383533.17          404920.77   
NE             370176.80          405274.21          443124.41   
PA             369877.20          410314.14          415781.46   
NC             368781.21          458386.62          486310.06   
CO             356024.83          412309.55          426198.32   
RI             349757.34          321878.90          353822.75   
AR             332116.35          441973.55          506680.88   
IL             330355.20          379220.85          395723.07   
VA             329679.09          326532.18          351749.96   
LA             327607.39          360921.79          373408.20   
SC             326637.72          347527.47          397129.95   
GA             325187.36          404745.24          430292.86   
NM             313188.27          365566.71          383304.47   
WI             312118.50          334234.29          374343.36   
KS             311466.99          330132.54          359850.94   
MI             309382.35          370899.31          408916.78   
DE             309249.64          332767.39          391146.20   
TN             289648.79          339160.74          368727.82   
OK             286416.50          314806.29          368126.05   
MD             271028.66          314727.16          378512.41   
CT             268620.23          237914.18          255290.80   
VI             268431.88          196494.48          169029.52   
MN             265116.04          310970.44          346755.82   
IA             263334.27          320628.67          368146.25   
WA             257405.48          311385.97          278178.14   
HI             254520.02          286415.35          286583.61   
CA             240849.09          248689.46          259502.51   
OR             233014.78          273247.90          235076.82   
NJ             232690.46          287274.95          283461.13   
AK             227011.06          233904.35          246871.20   
NY             171527.16          167906.13          189795.27   
GU             151643.55          156102.29          154547.92   
SD             144705.55          187814.75          197437.23   
ID             134169.21          136978.75          140242.01   
NH             128043.94          143346.83          160222.29   
MA             127168.92          134053.97          159915.94   
ME             103363.10          118908.71          160610.37   
VT              86547.72           88252.15           83014.41   
DC              59417.18           63455.64           68642.99   

       Total_rep_de_2017  Total_rep_de_2016  Total_rep_de_2015  \
State                                                            
WY             892322.33          856710.42          915502.80   
PR             556996.86          697742.95          804091.18   
FL             751406.71          766308.24   

Comparing both the tables,We can see that Wyoming tops the list in the 2020-2017.. other states seems to have reduced the direct emission better that Wyoming. 

In [10]:
df_de_it = df_de.groupby('IT_sectors')[['Total_rep_de_2020', 'Total_rep_de_2019',
       'Total_rep_de_2018', 'Total_rep_de_2017', 'Total_rep_de_2016',
       'Total_rep_de_2015', 'Total_rep_de_2014', 'Total_rep_de_2013',
       'Total_rep_de_2012', 'Total_rep_de_2011']].mean().apply(lambda x: round(x,2))
df_de_it.describe()

Total_rep_de_2020  Total_rep_de_2019  Total_rep_de_2018  \
count       6.200000e+01       6.200000e+01       6.200000e+01   
mean        1.491950e+06       1.568212e+06       1.617135e+06   
std         1.867757e+06       1.939223e+06       1.968377e+06   
min         2.804500e+02       1.892000e+02       3.068500e+02   
25%         9.116006e+04       1.077157e+05       1.214967e+05   
50%         4.784886e+05       7.082570e+05       7.488351e+05   
75%         2.207703e+06       2.268102e+06       2.214498e+06   
max         6.558090e+06       7.655594e+06       6.973092e+06   

       Total_rep_de_2017  Total_rep_de_2016  Total_rep_de_2015  \
count       6.200000e+01       6.200000e+01       6.200000e+01   
mean        1.569384e+06       1.583370e+06       1.645018e+06   
std         1.924594e+06       1.936530e+06       2.003390e+06   
min         3.581500e+02       2.950500e+02       4.840000e+01   
25%         1.492012e+05       1.624602e+05       1.754746e+05   
50%         7.198152e+05       6.634311e+05       7.037434e+05   
75%         2.169771e+06       2.375431e+06       2.397181e+06   
max         7.115795e+06       7.195194e+06       7.192878e+06   

       Total_rep_de_2014  Total_rep_de_2013  Total_rep_de_2012  \
count       6.200000e+01       6.200000e+01       6.200000e+01   
mean        1.651817e+06       1.687177e+06       1.665998e+06   
std         1.992219e+06       2.049031e+06       2.016930e+06   
min         2.807500e+02       1.575940e+03       2.787000e+02   
25%         1.974780e+05       2.074177e+05       2.524441e+05   
50%         7.386794e+05       7.457140e+05       7.589737e+05   
75%         2.227839e+06       2.258289e+06       2.254042e+06   
max         7.229060e+06       7.387407e+06       7.276498e+06   

       Total_rep_de_2011  
count       6.200000e+01  
mean        1.686357e+06  
std         2.080075e+06  
min         2.597000e+02  
25%         2.841495e+05  
50%         7.502238e+05  
75%         2.434158e+06  
max         7.303151e+06

In [11]:
df_de_it.sort_values(by='Total_rep_de_2020',ascending =False)

Total_rep_de_2020  \
IT_sectors                                                              
Petroleum Product Suppliers,Power Plants,Refine...         6558089.82   
Metals,Minerals                                            6173199.15   
Chemicals,Petroleum Product Suppliers,Power Pla...         5969593.13   
Metals,Power Plants                                        5963440.23   
Chemicals,Other,Petroleum Product Suppliers,Pow...         5106325.56   
...                                                               ...   
Chemicals,Industrial Gas Suppliers,Waste                     32761.46   
Import and Export of Equipment Containing Fluor...           30481.32   
Other,Petroleum and Natural Gas Systems                      14696.44   
Industrial Gas Suppliers,Other                               11121.51   
Natural Gas and Natural Gas Liquids Suppliers,O...             280.45   

                                                    Total_rep_de_2019  \
IT_sectors                                                              
Petroleum Product Suppliers,Power Plants,Refine...         7655594.26   
Metals,Minerals                                            6200941.83   
Chemicals,Petroleum Product Suppliers,Power Pla...         5917240.99   
Metals,Power Plants                                        5235337.92   
Chemicals,Other,Petroleum Product Suppliers,Pow...         4984461.77   
...                                                               ...   
Chemicals,Industrial Gas Suppliers,Waste                     42333.54   
Import and Export of Equipment Containing Fluor...           38044.08   
Other,Petroleum and Natural Gas Systems                      12696.30   
Industrial Gas Suppliers,Other                               11388.01   
Natural Gas and Natural Gas Liquids Suppliers,O...             189.20   

                                                    Total_rep_de_2018  \
IT_sectors                                                              
Petroleum Product Suppliers,Power Plants,Refine...         6973092.12   
Metals,Minerals                                            6360091.48   
Chemicals,Petroleum Product Suppliers,Power Pla...         5580514.16   
Metals,Power Plants                                        5379019.29   
Chemicals,Other,Petroleum Product Suppliers,Pow...         4812576.46   
...                                                               ...   
Chemicals,Industrial Gas Suppliers,Waste                     54078.74   
Import and Export of Equipment Containing Fluor...           45862.20   
Other,Petroleum and Natural Gas Systems                      27229.40   
Industrial Gas Suppliers,Other                               12234.50   
Natural Gas and Natural Gas Liquids Suppliers,O...             306.85   

                                                    Total_rep_de_2017  \
IT_sectors                                                              
Petroleum Product Suppliers,Power Plants,Refine...         7115794.78   
Metals,Minerals                                            6972176.48   
Chemicals,Petroleum Product Suppliers,Power Pla...         5787700.12   
Metals,Power Plants                                        3467216.61   
Chemicals,Other,Petroleum Product Suppliers,Pow...         4722108.68   
...                                                               ...   
Chemicals,Industrial Gas Suppliers,Waste                    103988.20   
Import and Export of Equipment Containing Fluor...           62335.20   
Other,Petroleum and Natural Gas Systems                      23020.89   
Industrial Gas Suppliers,Other                               12399.95   
Natural Gas and Natural Gas Liquids Suppliers,O...             358.15   

                                                    Total_rep_de_2016  \
IT_sectors                                                              
Petroleum Product Suppliers,Power Plants,Refine...         7195194.16   
Metals,Minerals                  

In [13]:
df_de_it_state = df_de[['State', 'IT_sectors']]

In [14]:
df_de_it_state[df_de_it_state.State == 'TX'].sort_values(by='IT_sectors')

State IT_sectors
Facility Id                 
1006890        TX  Chemicals
1003899        TX  Chemicals
1000346        TX  Chemicals
1003134        TX  Chemicals
1003781        TX  Chemicals
...           ...        ...
1006227        TX      Waste
1003001        TX      Waste
1002085        TX      Waste
1000372        TX      Waste
1004377        TX      Waste

[813 rows x 2 columns]

In [15]:
df_de_it_state.groupby('State').count().sort_values(by='IT_sectors', ascending=False)

IT_sectors
State            
TX            813
LA            414
CA            376
PA            280
OH            255
IL            233
MI            214
NY            204
IN            191
FL            173
AL            172
OK            169
GA            164
IA            152
KY            134
VA            133
MN            131
WI            129
NC            125
WV            118
TN            118
CO            112
MO            107
MS            106
KS            104
AR             99
SC             98
WA             90
AZ             89
NJ             87
NM             87
NE             72
MA             69
WY             68
UT             65
AK             63
ND             62
MD             58
OR             58
CT             40
NV             39
ID             37
SD             36
ME             28
HI             28
MT             28
PR             20
DE             19
NH             17
RI             10
GU              8
VT              5
VI              4
DC              4

In [16]:
df_de_it_st = df_de.groupby(['State','IT_sectors'])[['Total_rep_de_2020', 'Total_rep_de_2019',
       'Total_rep_de_2018', 'Total_rep_de_2017', 'Total_rep_de_2016',
       'Total_rep_de_2015', 'Total_rep_de_2014', 'Total_rep_de_2013',
       'Total_rep_de_2012', 'Total_rep_de_2011']].mean().apply(lambda x: round(x,2))


In [17]:
df_de_it_st.head()

Total_rep_de_2020  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             85459.44   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          475952.98   
      Other                                                        84518.85   
      Petroleum Product Suppliers,Refineries                      183045.13   
      Petroleum and Natural Gas Systems                           453520.61   

                                                          Total_rep_de_2019  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             85877.31   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          600981.03   
      Other                                                        76957.84   
      Petroleum Product Suppliers,Refineries                      202376.40   
      Petroleum and Natural Gas Systems                           445897.42   

                                                          Total_rep_de_2018  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             79770.61   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          695400.30   
      Other                                                       135133.68   
      Petroleum Product Suppliers,Refineries                      205750.48   
      Petroleum and Natural Gas Systems                           444387.77   

                                                          Total_rep_de_2017  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             73953.38   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          723807.79   
      Other                                                       137083.32   
      Petroleum Product Suppliers,Refineries                      199390.18   
      Petroleum and Natural Gas Systems                           442243.90   

                                                          Total_rep_de_2016  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             72235.88   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          589663.71   
      Other                                                       108366.96   
      Petroleum Product Suppliers,Refineries                      187904.77   
      Petroleum and Natural Gas Systems                           447281.86   

                                                          Total_rep_de_2015  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries             80653.72   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          604584.14   
      Other                                                       111860.71   
      Petroleum Product Suppliers,Refineries                      190899.49   
      Petroleum and Natural Gas Systems                           458422.66   

                                                          Total_rep_de_2014  \
State IT_sectors                                                              
AK    Chemicals,Petroleum Product Suppliers,Refineries            114751.11   
      Chemicals,Petroleum and Natural Gas Systems,Ref...          577309.02   
      Other                                                       115089.78   
      Petroleum Product Suppliers,Refineries                      193758.65   
      Petroleum and Natural Gas Systems                           438472.14   

                                                          Total_rep_de_2013  \
State IT_sectors                                                              
AK    Chemicals,Petrol

In [18]:
df_de_it_st.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 626 entries, ('AK', 'Chemicals,Petroleum Product Suppliers,Refineries') to ('WY', 'Waste')
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Total_rep_de_2020  626 non-null    float64
 1   Total_rep_de_2019  626 non-null    float64
 2   Total_rep_de_2018  626 non-null    float64
 3   Total_rep_de_2017  626 non-null    float64
 4   Total_rep_de_2016  626 non-null    float64
 5   Total_rep_de_2015  626 non-null    float64
 6   Total_rep_de_2014  626 non-null    float64
 7   Total_rep_de_2013  626 non-null    float64
 8   Total_rep_de_2012  626 non-null    float64
 9   Total_rep_de_2011  626 non-null    float64
dtypes: float64(10)
memory usage: 51.1+ KB


In [19]:
df_sf6.columns

Index(['Facility Id', 'FRD_id', 'Facility_Name', 'City', 'State', 'Zip_Code',
       'Address', 'County', 'Latitude', 'Longitude', 'Primary_NAICS_Code',
       'IT_subparts', 'Total_de_EE_2020', 'SF6_emissions_2020 ',
       'Total_de_EE_2019', 'SF6_emissions_2019', 'Total_de_EE_2018',
       'SF6_emissions_2018', 'Total_de_EE_2017', 'SF6_emissions_2017',
       'Total_de_EE_2016', 'SF6_emissions_2016', 'Total_de_EE_2015',
       'SF6_emissions_2015', 'Total_de_EE_2014', 'SF6_emissions_2014',
       'Total_de_EE_2013', 'SF6_emissions_2013', 'Total_de_EE_2012',
       'SF6_emissions_2012', 'Total_de_EE_2011', 'SF6_emissions_2011'],
      dtype='object')

In [17]:
df_sf6_it = df_sf6.groupby('Industry Type (subparts)')[['Total reported direct emissions from Electrical Equipment Use','Total reported direct emissions from Electrical Equipment Use_2019','Total reported direct emissions from Electrical Equipment Use_2018','Total reported direct emissions from Electrical Equipment Use_2017','Total reported direct emissions from Electrical Equipment Use_2016','Total reported direct emissions from Electrical Equipment Use_2015','Total reported direct emissions from Electrical Equipment Use_2014','Total reported direct emissions from Electrical Equipment Use_2013','Total reported direct emissions from Electrical Equipment Use_2012','Total reported direct emissions from Electrical Equipment Use_2011']].mean().apply(lambda x: round(x,2))


In [19]:
df_sf6_it.head()

Total reported direct emissions from Electrical Equipment Use  \
Industry Type (subparts)                                                                  
C,D,DD                                                                10.26               
C,D,DD,MM-REF,P,Y                                                   1158.24               
C,DD                                                                 133.00               
C,DD,TT,W-UNSTG,X                                                     29.64               
D,DD                                                                 381.33               

                          Total reported direct emissions from Electrical Equipment Use_2019  \
Industry Type (subparts)                                                                       
C,D,DD                                                                 0.00                    
C,D,DD,MM-REF,P,Y                                                   1454.64                    
C,DD                                                                   0.00                    
C,DD,TT,W-UNSTG,X                                                      0.00                    
D,DD                                                                 312.93                    

                          Total reported direct emissions from Electrical Equipment Use_2018  \
Industry Type (subparts)                                                                       
C,D,DD                                                              -511.86                    
C,D,DD,MM-REF,P,Y                                                   1083.00                    
C,DD                                                                 140.60                    
C,DD,TT,W-UNSTG,X                                                    567.72                    
D,DD                                                                  27.36                    

                          Total reported direct emissions from Electrical Equipment Use_2017  \
Industry Type (subparts)                                                                       
C,D,DD                                                                 9.12                    
C,D,DD,MM-REF,P,Y                                                    579.12                    
C,DD                                                                 129.20                    
C,DD,TT,W-UNSTG,X                                                   1643.88                    
D,DD                                                                1254.00                    

                          Total reported direct emissions from Electrical Equipment Use_2016  \
Industry Type (subparts)                                                                       
C,D,DD                                                                 0.00                    
C,D,DD,MM-REF,P,Y                                                    114.00                    
C,DD                                                                 155.80                    
C,DD,TT,W-UNSTG,X                                                      0.00                    
D,DD                                                                 856.14                    

                          Total reported direct emissions from Electrical Equipment Use_2015  \
Industry Type (subparts)                                                                       
C,D,DD                                                                 0.00                    
C,D,DD,MM-REF,P,Y                                                    428.64                    
C,DD                                                                 131.10                    
C,DD,TT,W-UNSTG,X                                                   1035.12                    
D,DD                                                                 290.70                    

                       

Sulfur hexafluoride (SF6) is a synthetic fluorinated compound with an extremely stable molecular structure. Because of its unique dielectric properties, electric utilities rely heavily on SF6 in electric power systems for voltage electrical insulation, current interruption, and arc quenching in the transmission and distribution of electricity. Yet, it is also the most potent greenhouse gas known to-date. Over a 100-year period, SF6 is 22,800 times more effective at trapping infrared radiation than an equivalent amount of carbon dioxide (CO2). SF6 is also a very stable chemical, with an atmospheric lifetime of 3,200 years. As the gas is emitted, it accumulates in the atmosphere in an essentially un-degraded state for many centuries. Thus, a relatively small amount of SF6 can have a significant impact on global climate change.

The electric power industry can reduce the nation’s SF6 emissions through cost-effective operational improvements and equipment upgrades. Through improvements in the leak rate of new equipment, refurbishing older equipment, and the use of more efficient operation and maintenance techniques, utilities often find economical solutions to reduce SF6 emissions. For more info, citation: https://www.epa.gov/eps-partnership/sulfur-hexafluoride-sf6-basics

In [85]:
df_sf6_it_st = df_sf6.groupby(['Reported State','Industry Type (subparts)'])[['Total reported direct emissions from Electrical Equipment Use','Total reported direct emissions from Electrical Equipment Use_2019','Total reported direct emissions from Electrical Equipment Use_2018','Total reported direct emissions from Electrical Equipment Use_2017','Total reported direct emissions from Electrical Equipment Use_2016','Total reported direct emissions from Electrical Equipment Use_2015','Total reported direct emissions from Electrical Equipment Use_2014','Total reported direct emissions from Electrical Equipment Use_2013','Total reported direct emissions from Electrical Equipment Use_2012','Total reported direct emissions from Electrical Equipment Use_2011']].mean().apply(lambda x: round(x,2))
df_sf6_it_st.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 44 entries, ('AL', 'DD') to ('WA', 'DD')
Data columns (total 10 columns):
 #   Column                                                              Non-Null Count  Dtype  
---  ------                                                              --------------  -----  
 0   Total reported direct emissions from Electrical Equipment Use       44 non-null     float64
 1   Total reported direct emissions from Electrical Equipment Use_2019  42 non-null     float64
 2   Total reported direct emissions from Electrical Equipment Use_2018  42 non-null     float64
 3   Total reported direct emissions from Electrical Equipment Use_2017  41 non-null     float64
 4   Total reported direct emissions from Electrical Equipment Use_2016  40 non-null     float64
 5   Total reported direct emissions from Electrical Equipment Use_2015  39 non-null     float64
 6   Total reported direct emissions from Electrical Equipment Use_2014  40 non-null     float64
 7 

In [27]:
df_sf6_it_state = df_sf6[['Reported State', 'Industry Type (subparts)']]

In [29]:
df_sf6_it_state.groupby('Reported State').count().sort_values(by='Industry Type (subparts)', ascending=False)

Industry Type (subparts)
Reported State                          
TX                                    14
CA                                    10
NY                                     6
WA                                     4
MI                                     4
FL                                     4
PA                                     4
MN                                     4
OH                                     3
IN                                     3
AZ                                     3
NV                                     2
MO                                     2
ND                                     2
KY                                     2
IA                                     2
MA                                     1
OR                                     1
VT                                     1
VA                                     1
UT                                     1
CO                                     1
TN                                     1
SC                                     1
PR                                     1
GA                                     1
OK                                     1
LA                                     1
HI                                     1
IL                                     1
NM                                     1
NJ                                     1
NE                                     1
KS                                     1
NC                                     1
AL                                     1

In [30]:
df_sf6_it_state.groupby('Industry Type (subparts)').count().sort_values(by='Reported State', ascending=False)

Reported State
Industry Type (subparts)                
DD                                    78
D,DD                                   4
C,DD                                   3
C,D,DD                                 2
C,D,DD,MM-REF,P,Y                      1
C,DD,TT,W-UNSTG,X                      1

In [32]:
df_sf6_it_st.sort_values(by='Total reported direct emissions from Electrical Equipment Use',ascending =False)

Total reported direct emissions from Electrical Equipment Use  \
Reported State Industry Type (subparts)                                                                  
NC             DD                                                                249121.92               
OH             DD                                                                 92007.88               
CO             DD                                                                 69218.52               
GA             DD                                                                 67909.80               
MN             DD                                                                 49673.79               
CA             DD                                                                 46470.96               
FL             DD                                                                 43358.76               
TN             DD                                                                 35547.48               
OK             DD                                                                 32902.68               
VA             DD                                                                 30973.80               
NM             DD                                                                 29047.20               
SC             DD                                                                 28037.16               
UT             DD                                                                 25203.12               
KY             DD                                                                 20884.80               
NJ             DD                                                                 18472.56               
NY             DD                                                                 16802.08               
AZ             DD                                                                 16429.68               
MA             DD                                                                 13816.80               
MI             DD                                                                 12520.05               
LA             DD                                                                 11625.72               
WA             DD                                                                 10676.10               
IA             DD                                                                 10560.96               
TX             DD                                                                 10044.02               
KS             DD                                                                  9993.24               
PR             DD                                                                  9742.44               
NE             DD                                                                  9560.04               
PA             DD                                                                  8532.52               
IL             DD                                                                  7644.84               
NV             DD                                                                  6321.30               
OR             DD                                                                  5538.12               
HI             DD                                                                  3600.12               
IN             DD                                                                  3345.90               
ND             DD                                                                  3125.88               
MO             DD                                                                  2960.58               
TX             D,DD                                                                1525.32               
IN             C,D,DD,MM-REF,P,Y                                                   1158.24               
VT           

In [37]:
df_sf6_it_state.sort_values(by='Reported State')

Reported State Industry Type (subparts)
62             AL                       DD
4              AZ                       DD
85             AZ                       DD
71             AZ                       DD
42             CA                       DD
..            ...                      ...
86             VT                       DD
74             WA                       DD
5              WA                       DD
9              WA                       DD
64             WA                       DD

[89 rows x 2 columns]

In [38]:
df_sf6_it_state.groupby(['Reported State','Industry Type (subparts)']).count().sort_values(by='Industry Type (subparts)', ascending=False)

Empty DataFrame
Columns: []
Index: [(AL, DD), (MO, DD), (AZ, DD), (NE, DD), (NJ, DD), (NM, DD), (NV, DD), (NY, DD), (OH, DD), (OK, DD), (OR, DD), (PA, DD), (PR, DD), (SC, DD), (TN, DD), (TX, DD), (UT, DD), (VA, DD), (VT, DD), (NC, DD), (ND, DD), (MN, DD), (IA, DD), (CA, DD), (CO, DD), (FL, DD), (GA, DD), (HI, DD), (MI, DD), (WA, DD), (IL, DD), (IN, DD), (KS, DD), (KY, DD), (LA, DD), (MA, DD), (TX, D,DD), (CA, D,DD), (TX, C,DD,TT,W-UNSTG,X), (TX, C,DD), (CA, C,DD), (IN, C,D,DD,MM-REF,P,Y), (PA, C,D,DD), (IA, C,D,DD)]

In [39]:
df_oog.columns

Index(['Facility Id', 'FRS Id', 'Facility Name', 'Basin', 'Reported City',
       'Reported State', 'Reported Zip Code', 'Reported Address',
       'Reported County', 'Reported Latitude', 'Reported Longitude',
       'Primary NAICS Code', 'Industry Type (subparts)',
       'Total reported emissions from Onshore Oil & Gas Production ',
       'CO2 emissions (non-biogenic) ', 'Methane (CH4) emissions ',
       'Nitrous Oxide (N2O) emissions ',
       'Total reported emissions from Onshore Oil & Gas Production _2019',
       'CO2 emissions (non-biogenic) _2019', 'Methane (CH4) emissions _2019',
       'Nitrous Oxide (N2O) emissions _2019',
       'Total reported emissions from Onshore Oil & Gas Production _2018',
       'CO2 emissions (non-biogenic) _2018', 'Methane (CH4) emissions _2018',
       'Nitrous Oxide (N2O) emissions _2018',
       'Total reported emissions from Onshore Oil & Gas Production _2017',
       'CO2 emissions (non-biogenic) _2017', 'Methane (CH4) emissions _2017',
   

In [45]:
df_oog_it_st = df_oog.groupby(['Reported State','Industry Type (subparts)'])[['Total reported emissions from Onshore Oil & Gas Production ','Total reported emissions from Onshore Oil & Gas Production _2019','Total reported emissions from Onshore Oil & Gas Production _2018','Total reported emissions from Onshore Oil & Gas Production _2017','Total reported emissions from Onshore Oil & Gas Production _2016','Total reported emissions from Onshore Oil & Gas Production _2015','Total reported emissions from Onshore Oil & Gas Production _2014','Total reported emissions from Onshore Oil & Gas Production _2013','Total reported emissions from Onshore Oil & Gas Production _2012','Total reported emissions from Onshore Oil & Gas Production _2011']].mean().apply(lambda x: round(x,2))


In [46]:
df_oog_it_st.head()

Total reported emissions from Onshore Oil & Gas Production   \
Reported State Industry Type (subparts)                                                                
AK             W-ONSH                                                            193788.34             
AR             UU,W-ONSH                                                          31360.80             
               W-ONSH                                                            725019.83             
CA             NN-FRAC,W-ONSH                                                     31463.09             
               W-ONSH                                                            153642.83             

                                         Total reported emissions from Onshore Oil & Gas Production _2019  \
Reported State Industry Type (subparts)                                                                     
AK             W-ONSH                                                            194756.38                  
AR             UU,W-ONSH                                                          32928.77                  
               W-ONSH                                                            690915.83                  
CA             NN-FRAC,W-ONSH                                                     26161.81                  
               W-ONSH                                                            181389.98                  

                                         Total reported emissions from Onshore Oil & Gas Production _2018  \
Reported State Industry Type (subparts)                                                                     
AK             W-ONSH                                                            205199.25                  
AR             UU,W-ONSH                                                          31222.60                  
               W-ONSH                                                            666601.28                  
CA             NN-FRAC,W-ONSH                                                     26581.46                  
               W-ONSH                                                            174891.11                  

                                         Total reported emissions from Onshore Oil & Gas Production _2017  \
Reported State Industry Type (subparts)                                                                     
AK             W-ONSH                                                            163625.17                  
AR             UU,W-ONSH                                                          27933.25                  
               W-ONSH                                                            261873.53                  
CA             NN-FRAC,W-ONSH                                                     27277.41                  
               W-ONSH                                                            168409.82                  

                                         Total reported emissions from Onshore Oil & Gas Production _2016  \
Reported State Industry Type (subparts)                                                                     
AK             W-ONSH                                                            179725.83                  
AR             UU,W-ONSH                                                          31319.84                  
               W-ONSH                                                            261063.78                  
CA             NN-FRAC,W-ONSH                                                          NaN                  
               W-ONSH                                                             98141.67                  

                                         Total reported emissions from Onshore Oil & Gas Production _2015  \
Reported State Industry Type (subparts)                                                                     
AK             W-ONSH                 

In [48]:
df_oog_it_st.sort_values(by='Total reported emissions from Onshore Oil & Gas Production ', ascending=False)

Total reported emissions from Onshore Oil & Gas Production   \
Reported State Industry Type (subparts)                                                                
OK             C,W-ONSH                                                         1275834.23             
AR             W-ONSH                                                            725019.83             
ND             W-ONSH                                                            470553.06             
OH             W-ONSH                                                            306758.07             
TX             W-ONSH                                                            211828.98             
NM             W-ONSH                                                            210710.64             
CO             W-ONSH                                                            206331.42             
OK             W-ONSH                                                            199689.36             
AK             W-ONSH                                                            193788.34             
CA             W-ONSH                                                            153642.83             
UT             W-ONSH                                                            138898.70             
PA             W-ONSH                                                            129120.73             
TX             UU,W-ONSH                                                          97578.95             
WY             W-ONSH                                                             97324.77             
MI             W-ONSH                                                             77063.54             
WV             W-ONSH                                                             76350.07             
LA             W-ONSH                                                             62512.22             
CO             UU,W-ONSH                                                          51563.82             
MS             W-ONSH                                                             49101.21             
MT             W-ONSH                                                             44226.69             
CA             NN-FRAC,W-ONSH                                                     31463.09             
AR             UU,W-ONSH                                                          31360.80             
MI             RR (RPT),W-ONSH                                                    29252.35             

                                         Total reported emissions from Onshore Oil & Gas Production _2019  \
Reported State Industry Type (subparts)                                                                     
OK             C,W-ONSH                                                         1692275.99                  
AR             W-ONSH                                                            690915.83                  
ND             W-ONSH                                                            833563.94                  
OH             W-ONSH                                                            414288.92                  
TX             W-ONSH                                                            263475.95                  
NM             W-ONSH                                                            276464.17                  
CO             W-ONSH                                                            314738.15                  
OK             W-ONSH                                                            266860.14                  
AK             W-ONSH                                                            194756.38                  
CA             W-ONSH                                                            181389.98                  
UT             W-ONSH                                                            151119.61                  
PA             W-ONSH  

In [49]:
df_ldc.columns

Index(['Facility Id', 'FRS Id', 'Facility Name', 'State where Emissions Occur',
       'Reported City', 'Reported State', 'Reported Zip Code',
       'Reported Address', 'Reported County', 'Reported Latitude',
       'Reported Longitude', 'Primary NAICS Code', 'Industry Type (subparts)',
       'Total reported direct emissions from Local Distribution Companies',
       'Total reported direct emissions from Local Distribution Companies_2019',
       'Total reported direct emissions from Local Distribution Companies_2018',
       'Total reported direct emissions from Local Distribution Companies_2017',
       'Total reported direct emissions from Local Distribution Companies_2016',
       'Total reported direct emissions from Local Distribution Companies_2015',
       'Total reported direct emissions from Local Distribution Companies_2014',
       'Total reported direct emissions from Local Distribution Companies_2013',
       'Total reported direct emissions from Local Distribution Comp

In [50]:
df_ldc_it_st = df_ldc.groupby(['Reported State','Industry Type (subparts)'])[['Total reported direct emissions from Local Distribution Companies',
       'Total reported direct emissions from Local Distribution Companies_2019',
       'Total reported direct emissions from Local Distribution Companies_2018',
       'Total reported direct emissions from Local Distribution Companies_2017',
       'Total reported direct emissions from Local Distribution Companies_2016',
       'Total reported direct emissions from Local Distribution Companies_2015',
       'Total reported direct emissions from Local Distribution Companies_2014',
       'Total reported direct emissions from Local Distribution Companies_2013',
       'Total reported direct emissions from Local Distribution Companies_2012',
       'Total reported direct emissions from Local Distribution Companies_2011']].mean().apply(lambda x: round(x,2))

In [51]:
df_ldc_it_st.head()

Total reported direct emissions from Local Distribution Companies  \
Reported State Industry Type (subparts)                                                                      
AL             NN-LDC,W-LDC                                                       37914.10                   
AR             NN-LDC,W-LDC                                                      164066.75                   
AZ             NN-LDC,W-LDC                                                       10825.92                   
CA             NN-LDC,W-LDC                                                      548482.12                   
CO             NN-LDC,W-LDC                                                      110734.68                   

                                         Total reported direct emissions from Local Distribution Companies_2019  \
Reported State Industry Type (subparts)                                                                           
AL             NN-LDC,W-LDC                                                       39847.33                        
AR             NN-LDC,W-LDC                                                      104075.80                        
AZ             NN-LDC,W-LDC                                                       10642.30                        
CA             NN-LDC,W-LDC                                                      548295.63                        
CO             NN-LDC,W-LDC                                                      185908.80                        

                                         Total reported direct emissions from Local Distribution Companies_2018  \
Reported State Industry Type (subparts)                                                                           
AL             NN-LDC,W-LDC                                                       41918.30                        
AR             NN-LDC,W-LDC                                                       97976.25                        
AZ             NN-LDC,W-LDC                                                       10572.00                        
CA             NN-LDC,W-LDC                                                      549075.10                        
CO             NN-LDC,W-LDC                                                      184926.45                        

                                         Total reported direct emissions from Local Distribution Companies_2017  \
Reported State Industry Type (subparts)                                                                           
AL             NN-LDC,W-LDC                                                       44683.78                        
AR             NN-LDC,W-LDC                                                       65689.00                        
AZ             NN-LDC,W-LDC                                                       10460.60                        
CA             NN-LDC,W-LDC                                                      549689.63                        
CO             NN-LDC,W-LDC                                                      188475.30                        

                                         Total reported direct emissions from Local Distribution Companies_2016  \
Reported State Industry Type (subparts)                                                                           
AL             NN-LDC,W-LDC                                                       45969.70                        
AR             NN-LDC,W-LDC                                                       61738.55                        
AZ             NN-LDC,W-LDC                                                       10406.50                        
CA             NN-LDC,W-LDC                                                      552440.63                        
CO             NN-LDC,W-LDC                                                      192661.25                        

                                         Total

In [52]:
df_ldc_it_state = df_ldc[['Reported State', 'Industry Type (subparts)']]

In [54]:
df_ldc_it_state.groupby('Reported State').count()

Industry Type (subparts)
Reported State                          
AL                                     5
AR                                     1
AZ                                     2
CA                                     3
CO                                     2
CT                                     4
DE                                     2
FL                                     5
GA                                     1
IA                                     6
ID                                     1
IL                                     5
IN                                    11
KS                                     2
KY                                     3
LA                                     2
MA                                     6
MD                                     3
MI                                     3
MN                                     3
MO                                     5
MS                                     1
MT                                     3
NC                                     5
NE                                     2
NJ                                     4
NM                                     2
NV                                     3
NY                                     8
OH                                     3
OK                                     1
OR                                     2
PA                                     6
RI                                     1
SC                                     3
TN                                     8
TX                                    18
UT                                     2
VA                                     6
VT                                     1
WA                                     3
WI                                     5
WV                                     2

In [55]:
df_ldc_it_state.groupby('Industry Type (subparts)').count()

Reported State
Industry Type (subparts)                
NN-LDC,W-LDC                         164

In [56]:
df_ldc_it_st.sort_values(by='Total reported direct emissions from Local Distribution Companies',ascending=False)

Total reported direct emissions from Local Distribution Companies  \
Reported State Industry Type (subparts)                                                                      
CA             NN-LDC,W-LDC                                                      548482.12                   
MI             NN-LDC,W-LDC                                                      229807.38                   
OH             NN-LDC,W-LDC                                                      208308.80                   
NJ             NN-LDC,W-LDC                                                      177008.61                   
GA             NN-LDC,W-LDC                                                      165161.15                   
AR             NN-LDC,W-LDC                                                      164066.75                   
OK             NN-LDC,W-LDC                                                      149571.85                   
PA             NN-LDC,W-LDC                                                      148489.66                   
RI             NN-LDC,W-LDC                                                      137393.40                   
NY             NN-LDC,W-LDC                                                      134802.81                   
CO             NN-LDC,W-LDC                                                      110734.68                   
WV             NN-LDC,W-LDC                                                      104260.42                   
MA             NN-LDC,W-LDC                                                      100909.54                   
TX             NN-LDC,W-LDC                                                       80444.26                   
IL             NN-LDC,W-LDC                                                       76699.79                   
MD             NN-LDC,W-LDC                                                       64973.82                   
IN             NN-LDC,W-LDC                                                       63465.03                   
NE             NN-LDC,W-LDC                                                       55295.30                   
NC             NN-LDC,W-LDC                                                       52776.63                   
MN             NN-LDC,W-LDC                                                       50824.11                   
NV             NN-LDC,W-LDC                                                       50544.97                   
CT             NN-LDC,W-LDC                                                       50440.75                   
KS             NN-LDC,W-LDC                                                       50142.35                   
UT             NN-LDC,W-LDC                                                       48840.15                   
VA             NN-LDC,W-LDC                                                       40118.55                   
AL             NN-LDC,W-LDC                                                       37914.10                   
WI             NN-LDC,W-LDC                                                       35612.38                   
WA             NN-LDC,W-LDC                                                       34626.00                   
MO             NN-LDC,W-LDC                                                       33325.10                   
OR             NN-LDC,W-LDC                                                       33190.20                   
ID             NN-LDC,W-LDC                                                       32433.00                   
FL             NN-LDC,W-LDC                                                       29033.78                   
NM             NN-LDC,W-LDC                                                       27824.75                   
IA             NN-LDC,W-LDC                                                       19270.14                   
SC             NN-LDC,W-LDC                                                      

In [80]:
df_ldc.isna().count()

Facility Id                                                               164
FRS Id                                                                    164
Facility Name                                                             164
State where Emissions Occur                                               164
Reported City                                                             164
Reported State                                                            164
Reported Zip Code                                                         164
Reported Address                                                          164
Reported County                                                           164
Reported Latitude                                                         164
Reported Longitude                                                        164
Primary NAICS Code                                                        164
Industry Type (subparts)                                        

In [82]:
df_ldc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 23 columns):
 #   Column                                                                  Non-Null Count  Dtype  
---  ------                                                                  --------------  -----  
 0   Facility Id                                                             164 non-null    int64  
 1   FRS Id                                                                  161 non-null    float64
 2   Facility Name                                                           164 non-null    object 
 3   State where Emissions Occur                                             163 non-null    object 
 4   Reported City                                                           164 non-null    object 
 5   Reported State                                                          164 non-null    object 
 6   Reported Zip Code                                                       164 non-nu

In [68]:
ldc_scale = scale(df_ldc_it_st)
df_ldc_columns= df_ldc_it_st.columns
df_ldc_scaled = pd.DataFrame(ldc_scale, columns = df_ldc_columns)

In [69]:
df_ldc_scaled.sort_values(by='Total reported direct emissions from Local Distribution Companies',ascending=False)

Total reported direct emissions from Local Distribution Companies  \
3                                            5.028471                   
18                                           1.608189                   
29                                           1.377448                   
25                                           1.041508                   
8                                            0.914351                   
1                                            0.902605                   
30                                           0.747034                   
32                                           0.735419                   
33                                           0.616325                   
28                                           0.588521                   
4                                            0.330201                   
42                                           0.260714                   
16                                           0.224750                   
36                                           0.005099                   
11                                          -0.035089                   
17                                          -0.160942                   
12                                          -0.177136                   
24                                          -0.264820                   
23                                          -0.291853                   
19                                          -0.312809                   
27                                          -0.315805                   
5                                           -0.316923                   
13                                          -0.320126                   
37                                          -0.334102                   
38                                          -0.427710                   
0                                           -0.451370                   
41                                          -0.476074                   
40                                          -0.486660                   
20                                          -0.500623                   
31                                          -0.502070                   
10                                          -0.510197                   
7                                           -0.546681                   
26                                          -0.559657                   
9                                           -0.651472                   
34                                          -0.688095                   
6                                           -0.690906                   
15                                          -0.698054                   
22                                          -0.727944                   
2                                           -0.742103                   
14                                          -0.768877                   
35                                          -0.782506                   
39                                          -0.814643                   
21                                          -0.825392                   

    Total reported direct emissions from Local Distribution Companies_2019  \
3                                            4.938638                        
18                                           1.687238                        
29                                           1.383238                        
25                                           1.146613                        
8                                            0.764455                        
1                                            0.229663                        
30                                           0.754107                        
32                                           0.747619                        
33                                           0.637037                        
28         

In [70]:
df_ldc_scaled.mean()

Total reported direct emissions from Local Distribution Companies         6.454785e-17
Total reported direct emissions from Local Distribution Companies_2019   -7.745742e-17
Total reported direct emissions from Local Distribution Companies_2018   -7.229359e-17
Total reported direct emissions from Local Distribution Companies_2017   -1.632292e-16
Total reported direct emissions from Local Distribution Companies_2016   -5.163828e-17
Total reported direct emissions from Local Distribution Companies_2015   -4.905637e-17
Total reported direct emissions from Local Distribution Companies_2014   -5.022437e-17
Total reported direct emissions from Local Distribution Companies_2013    1.876806e-16
Total reported direct emissions from Local Distribution Companies_2012   -2.581914e-17
Total reported direct emissions from Local Distribution Companies_2011    1.910616e-16
dtype: float64

In [72]:
df_ldc_scaled.std(ddof=0)

Total reported direct emissions from Local Distribution Companies         1.0
Total reported direct emissions from Local Distribution Companies_2019    1.0
Total reported direct emissions from Local Distribution Companies_2018    1.0
Total reported direct emissions from Local Distribution Companies_2017    1.0
Total reported direct emissions from Local Distribution Companies_2016    1.0
Total reported direct emissions from Local Distribution Companies_2015    1.0
Total reported direct emissions from Local Distribution Companies_2014    1.0
Total reported direct emissions from Local Distribution Companies_2013    1.0
Total reported direct emissions from Local Distribution Companies_2012    1.0
Total reported direct emissions from Local Distribution Companies_2011    1.0
dtype: float64

In [75]:
ldc_pca = PCA().fit(df_ldc_scaled)

ValueError: Input contains NaN, infinity or a value too large for dtype('float64').

In [76]:
df_ldc_it_st.isna().count()

Total reported direct emissions from Local Distribution Companies         43
Total reported direct emissions from Local Distribution Companies_2019    43
Total reported direct emissions from Local Distribution Companies_2018    43
Total reported direct emissions from Local Distribution Companies_2017    43
Total reported direct emissions from Local Distribution Companies_2016    43
Total reported direct emissions from Local Distribution Companies_2015    43
Total reported direct emissions from Local Distribution Companies_2014    43
Total reported direct emissions from Local Distribution Companies_2013    43
Total reported direct emissions from Local Distribution Companies_2012    43
Total reported direct emissions from Local Distribution Companies_2011    43
dtype: int64

In [ ]:
df_ldc_dropped_na.